sdk/keyvault/Azure.Security.KeyVault.Keys/src/Model.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Azure.Core;
+using System;
+using System.IO;
+using System.Text.Json;
+
+namespace Azure.Security.KeyVault.Keys
+{
+    public abstract class Model
```

Comment:
We need to avoid having base classes that are used for code sharing only.

Reply:
Could you expand more on why?

Reply:
We don't have a public API surface that benefits from having this type (nothing public takes or returns `Model`) and we only use it for sharing internal code so why have it?

Reply:
@pakrym are you suggesting removing the `Model` type completely or making it `internal`/ hiding it in some way?

Reply:
The compiler won't allow it to be internal, changing it to an internal interface would work though.

Reply:
Does this require a broader discussion about having a `Model` type? If that is the case, I would like to address both Keys and Secrets from the `Model` perspective at the same time, in a separate PR.

Reply:
I talked to Pavel about it yesterday and have an issue for it as the same happens for Page class.
See issue https://github.com/Azure/azure-sdk-for-net/issues/6477